crawl-ref/source/macro.cc

```diff
@@ -56,6 +56,10 @@
 #include "unicode.h"
 #include "version.h"
 
+// XXX: we might not want to use an std::deque here as they can waste a
```

Comment:
4096 bytes seems overkill certainly, but also probably not an issue for a one-off global.  Maybe take a note or create an issue and leave this for another PR?

Reply:
Each entry in the keymaps contains two deques so it's about 8 kB of wasted space per keymap entry with Clang (1 kB with GCC). It looks like keymaps only holds user defined keymaps and not any of the default key mappings so this isn't as bad as I initially thought.

I will move this to another issue or PR as it doesn't belong here